### PR TITLE
feat: First Time Onboarding

### PR DESCRIPTION
**Pull request by markphelps**
This is a follow up to #2730 

It has the following behaviour

- First time user runs Flipt, there will be no local storage `user: { completedOnboarding }` value set, so it will show the new `Onboarding` page with a button to say 'Completed'
- Once user clicks this button, it will set the above mentioned value in local storage and redirect user to `/flags` which was the previous default page
- User can still access `Onboarding` page from `Support`, but no `Complete` button will show

This PR also adds a new Button variant `soft` and changes the Button component to use `variant` instead of `primary`

It also fixes an issue with memorization and redux

### CODE REVIEW

File: ui/src/App.tsx [hunk 1]
```diff
@@ -22,17 +22,15 @@ const Console = loadable(() => import('./app/console/Console'));
...
 const Tokens = loadable(() => import('./app/tokens/Tokens'));
 
-const namespacesRoutes = [
+const namespacedRoutes = [
   {
-    element: <Flags />,
-    handle: {
-      namespaced: true
-    },
+    element: <FirstTimeOnboarding />,
```
Comment (erka) [inline ui/src/App.tsx:33]: I would suggest to drop `FirstTimeOnboarding` and just replace it with
```suggestion
    element: <Onboarding firstTime={true} />,
    loader: () => {
      const state = store.getState();
      if (state?.user.completedOnboarding) {
        return redirect('/flags');
      }
      return null;
    }

**Review by erka — COMMENTED** [review]
It looks great!

I have only one comment and suggestion. 
I have doubts about naming of `user` store.  It could confuse a bit in a future. `self` in session is actually user information and having `user` in store with non user information... Sadly I don't have any suggestion about better naming right now.

**Comment by markphelps** [comment]
> It looks great!
> 
> I have only one comment and suggestion. I have doubts about naming of `user` store. It could confuse a bit in a future. `self` in session is actually user information and having `user` in store with non user information... Sadly I don't have any suggestion about better naming right now.

Good idea. I renamed it to `eventStore` as its an event the user performs in the UI. Happy to think of other names as well though

File: ui/src/App.tsx (hunk 1 above)
Comment (markphelps) [reply ui/src/App.tsx]: i like it

**Review by erka — APPROVED** [review]
perfect!

**Pull request merged**